### DISCUSSION TITLE
Typescript usage

### DISCUSSION THREAD

**Discussion by martinlombana** [c-smile/sciter-js-sdk#79] [Q&A] [answered]
Regarding typescript usage, has anyone tried it? 

Should I use https://github.com/justjake/quickjs-emscripten

Or what do you recommend? I prefer typescript to regular JS as I do most things in Typescript, but I don't know if it will pose some problems here or if there are any limitations @c-smile .

Thanks in advance for the insight,

**Comment by c-smile** [comment, marked as answer]
Typescript is just a preprocessor that compiles .ts files to .js files.  So it has nothing with Sciter per se - you will load those .js files in Sciter as usual.

If you want standalone compiler then check :

> Typescript compiler compiled with QuickJS: [quickjs-typescript-4.0.0-linux-x86.tar.xz](https://bellard.org/quickjs/quickjs-typescript-4.0.0-linux-x86.tar.xz)

( link taken from https://bellard.org/quickjs/ )

In principle that compiler should work in Sciter "as it is" so you can load .ts files in Sciter directly. If that makes sense of course.

**Reply by martinlombana** [reply]
Yes, I knew that it is a preprocessor... I was asking more about compatibility issues between those bindings and the embedded version of quick-js interpreter. So I deduce that the embeded version of quick-js is always the official one?

And yes, that would be nice, of course, loading TS files directly. Although the link that you sent seems to be for linux only and there is no windows alternative, it seems.

**Reply by c-smile** [reply]
> the embeded version of quick-js is always the official one?

Yes, superset of official QuickJS

> the link that you sent seems to be for linux only

It contains .ts compiler sources + patch of .ts compiler to work inside Quick.JS
